comp/dogstatsd/server/parse.go

```diff
@@ -206,15 +202,18 @@ func (p *parser) parseMetricSample(message []byte) (dogstatsdMetricSample, error
 				return dogstatsdMetricSample{}, fmt.Errorf("dogstatsd timestamp should be > 0")
 			}
 			timestamp = time.Unix(ts, 0)
-		// container ID
+		// local data
 		case p.dsdOriginEnabled && bytes.HasPrefix(optionalField, localDataPrefix):
-			containerID = p.resolveContainerIDFromLocalData(optionalField)
+			localData, err = p.resolveLocalData(optionalField[len(localDataPrefix):])
+			if err == nil {
+				containerID = localData.ContainerID
+			}
 		// external data
 		case p.dsdOriginEnabled && bytes.HasPrefix(optionalField, externalDataPrefix):
 			rawExternalData := string(optionalField[len(externalDataPrefix):])
 			externalData, err = origindetection.ParseExternalData(rawExternalData)
 			if err != nil {
-				return dogstatsdMetricSample{}, fmt.Errorf("failed to parse OriginInfo.ExternalData %s: %v", rawExternalData, err)
+				return dogstatsdMetricSample{}, fmt.Errorf("failed to parse e: field containing External Data %s: %v", rawExternalData, err)
```

Comment:
Since `rawExternalData` was received from the network and can contain arbitrary data, let's quote it to avoid it [breaking](https://go.dev/play/p/v8WL9O8PtHY) the logs:

```suggestion
				return dogstatsdMetricSample{}, fmt.Errorf("failed to parse e: field containing External Data %q: %v", rawExternalData, err)
```